test/test_other.py

```diff
@@ -10551,10 +10585,23 @@ def test_assertions_on_ready_promise(self):
     ''')
     self.run_process([EMCC, test_file('hello_world.c'), '-sMODULARIZE', '-sASSERTIONS', '--extern-post-js', 'test.js'])
     # A return code of 7 is from the unhandled Promise rejection
-    out = self.run_js('a.out.js', assert_returncode=7)
+    out = self.run_js('a.out.js', assert_returncode=1)
```

Comment:
Please double check that you are ok with this change in error code @sbc100 - otherwise I think this is good to go.